pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -28,43 +28,45 @@ import (
 )
 
 type cyclonedxParser struct {
-	doc         *processor.Document
-	rootPackage parentPackages
+	doc           *processor.Document
+	rootComponent component
+	pkgMap        map[string]*component
```

Comment:
I think as to @pxp928 's comment we would want a parser test here! That should round up the PR, the rest LGTM.